### PR TITLE
Replace Java ExtractorCLI with C# ExtractionLauncher

### LINKED ISSUE #696: Replace Java ExtractorCLI with equivalent C# service
The purpose of the ExtractorCLI tool is to kick-start an extraction request by reading a CSV file and some CLI arguments, and sending a few RabbitMQ messages.

Currently this is implemented in Java which, although it is functional, has a few maintainability issues:
-   Duplication of message classes between C#/Java which require to be updated in sync
-   Requires a separate application to be deployed instead of being bundled in the single `smi` script
-   Our developers are less familiar with the Java ecosystem, and so that part of the repo is more susceptible to bit-rot

I don't see any strong reason why we shouldn't replace the Java service with an equivalent C# implementation

### PR DESCRIPTION
**Pull request by rkm**
## Proposed Changes

Replaces the Java `ExtractorCLI` service with the C# equivalent. Runnable via `./smi extraction-launcher ...` with the same options as before.

## Types of changes

What types of changes does your code introduce? Tick all that apply.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [x] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
    -   Existing scripts and documentation
    -   Change `ExtractorClOptions` -> `ExtractionLauncherOptions` in YAML configs
- [ ] Documentation-Only Update (if none of the other choices apply)
  - In this case, ensure that the message of the head commit from the source branch is prefixed with `[skip ci]`

## Checklist

By opening this PR, I confirm that I have:

- [x] Reviewed the [contributing](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTING.md) guidelines for this repository
- [x] Ensured that the PR branch is in sync with the target branch (i.e. it is automatically merge-able)
- [x] Updated any relevant API documentation
- [x] Created or updated any tests if relevant
- [x] Created a [news](https://github.com/SMI/SmiServices/blob/master/news/README.md) file
    -   NOTE: This ***must*** include any changes to any of the following files: default.yaml, any of the RabbitMQ server configurations, GlobalOptions.cs
- [x] Listed myself in the [CONTRIBUTORS](https://github.com/SMI/SmiServices/blob/master/CONTRIBUTORS.md) file 🚀
- [x] Requested a review by one of the repository maintainers

## Issues

-   Closes #696

### CODE REVIEW

File: src/applications/Applications.ExtractionLauncher/ExtractionMessageSender.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                KeyTag = extractionKey.ToString(),
+
+                Modalities = modalitiesString,
+
+                // NOTE(rkm 2021-04-01) Set below
+                ExtractionIdentifiers = null,
+            };
+
+            var ermList = new List<ExtractionRequestMessage>();
+            for (var i = 0; i <= (idList.Count / _maxIdentifiersPerMessage); ++i)
```
Comment (tznind) [inline src/applications/Applications.ExtractionLauncher/ExtractionMessageSender.cs:93]: Might be more maintainable:

```csharp

/// <summary>
/// Break a list of items into chunks of a specific size
/// </summary>
public static IEnumerable<IEnumerable<T>> Chunk<T>(this IEnumerable<T> source, int chunksize)
{
    while (source.Any())
    {
        yield return source.Take(chunksize);
        source = source.Skip(chunksize);
    }
}
```
https://stackoverflow.com/a/6362642

File: src/applications/Applications.SmiRunner/ServiceVerbs.cs [hunk 2]
```diff
@@ -18,6 +18,9 @@ public abstract class ApplicationVerbBase : VerbBase
     [Verb("dicom-directory-processor", HelpText = "Queue dicom files on disk for ETL")]
     public sealed class DicomDirectoryProcessorVerb : ApplicationVerbBase { }
 
+    [Verb("extraction-launcher", HelpText = "Launch an extraction job")]
```
Comment (tznind) [inline src/applications/Applications.SmiRunner/ServiceVerbs.cs:21]: Maybe just 'extract' i.e. 

```bash
./smi extract c:\myidentifiers
```

**Review by tznind — CHANGES_REQUESTED** [review]

File: src/applications/Applications.ExtractionLauncher/ExtractionMessageSender.cs (hunk 1 above)
Comment (rkm) [reply src/applications/Applications.ExtractionLauncher/ExtractionMessageSender.cs:93]: Good catch - will change to that

File: src/applications/Applications.SmiRunner/ServiceVerbs.cs (hunk 2 above)
Comment (rkm) [reply src/applications/Applications.SmiRunner/ServiceVerbs.cs:21]: How about `extract-images` ? I think `extract` on its own is a bit ambiguous.

Comment (tznind) [reply src/applications/Applications.SmiRunner/ServiceVerbs.cs:21]: I'd be fine with `extract-images`

**Comment by rkm** [comment]
Will fixup that test after lunch, so I'd hold-off on the review until then

**Review by tznind — APPROVED** [review]

**Pull request merged**